ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/git/GitServiceClientImpl.java

```diff
@@ -445,69 +419,60 @@ public void push(DevMachine devMachine,
                                   .send(dtoUnmarshallerFactory.newUnmarshaller(PushResponse.class));
     }
 
-    /** {@inheritDoc} */
     @Override
     public void remoteList(DevMachine devMachine,
                            ProjectConfigDto project,
                            @Nullable String remoteName,
                            boolean verbose,
                            AsyncRequestCallback<List<Remote>> callback) {
-        RemoteListRequest remoteListRequest = dtoFactory.createDto(RemoteListRequest.class).withVerbose(verbose);
-        if (remoteName != null) {
-            remoteListRequest.setRemote(remoteName);
-        }
-        String url = appContext.getDevMachine().getWsAgentBaseUrl() + REMOTE_LIST + "?projectPath=" + project.getPath();
-        asyncRequestFactory.createPostRequest(url, remoteListRequest).loader(loader).send(callback);
+        String params = "?projectPath=" + project.getPath() + (remoteName != null ? "&remoteName=" + remoteName : "") +
+                        "&verbose=" + String.valueOf(verbose);
+        String url = appContext.getDevMachine().getWsAgentBaseUrl() + REMOTE + params;
+        asyncRequestFactory.createGetRequest(url).loader(loader).send(callback);
     }
 
-    /** {@inheritDoc} */
     @Override
     public Promise<List<Remote>> remoteList(DevMachine devMachine, ProjectConfig project, @Nullable String remoteName, boolean verbose) {
-        RemoteListRequest remoteListRequest = dtoFactory.createDto(RemoteListRequest.class).withVerbose(verbose);
-        if (remoteName != null) {
-            remoteListRequest.setRemote(remoteName);
-        }
-        String url = devMachine.getWsAgentBaseUrl() + REMOTE_LIST + "?projectPath=" + project.getPath();
-        return asyncRequestFactory.createPostRequest(url, remoteListRequest)
+        String params = "?projectPath=" + project.getPath() + (remoteName != null ? "&remoteName=" + remoteName : "") +
```

Comment:
I would move this chech `(remoteName != null ? "&remoteName=" + remoteName : "")` to if
```
if (remoteName != null) {
  params += &remoteName=" + remoteName;
}
```